packages/devtools_app/test/memory/control/settings_dialog_test.dart

```diff
@@ -0,0 +1,138 @@
+  late PreferencesController preferencesController;
+
+  void _setUpServiceManagerForMemory() {
+    // Load canned data testHeapSampleData.
+    final memoryJson =
+        SamplesMemoryJson.decode(argJsonString: testHeapSampleData);
+    final allocationJson =
+        AllocationMemoryJson.decode(argJsonString: testAllocationData);
+
+    fakeServiceManager = FakeServiceManager(
+      service: FakeServiceManager.createFakeService(
+        memoryData: memoryJson,
+        allocationData: allocationJson,
+      ),
+    );
+    when(fakeServiceManager.connectedApp!.isDartWebAppNow).thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isFlutterAppNow).thenReturn(true);
+    when(fakeServiceManager.connectedApp!.isDartCliAppNow).thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isDebugFlutterAppNow)
+        .thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isDartWebApp)
+        .thenAnswer((_) => Future.value(false));
+    setGlobal(ServiceConnectionManager, fakeServiceManager);
+
+    controller.offline.value = true;
+    controller.memoryTimeline.offlineData.clear();
+    controller.memoryTimeline.offlineData.addAll(memoryJson.data);
+  }
+
+  Future<void> pumpMemoryScreen(
+    WidgetTester tester, {
+    MemoryController? memoryController,
+  }) async {
+    await tester.pumpWidget(
+      wrapWithControllers(
+        const MemoryBody(),
+        memory: controller = memoryController ?? MemoryController(),
+      ),
+    );
+
+    // Delay to ensure the memory profiler has collected data.
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+    expect(find.byType(MemoryBody), findsOneWidget);
+  }
+
+  // Set a wide enough screen width that we do not run into overflow.
+  const windowSize = Size(2225.0, 1000.0);
+
+  group('MemoryScreen', () {
+    setUp(() async {
+      await ensureInspectorDependencies();
+      setGlobal(OfflineModeController, OfflineModeController());
+      fakeServiceManager = FakeServiceManager();
+      when(fakeServiceManager.connectedApp!.isDartWebAppNow).thenReturn(false);
+      when(fakeServiceManager.connectedApp!.isDebugFlutterAppNow)
+          .thenReturn(false);
+      when(fakeServiceManager.vm.operatingSystem).thenReturn('android');
+      when(fakeServiceManager.connectedApp!.isDartWebApp)
+          .thenAnswer((_) => Future.value(false));
+      when(fakeServiceManager.errorBadgeManager.errorCountNotifier('memory'))
+          .thenReturn(ValueNotifier<int>(0));
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+      setGlobal(IdeTheme, IdeTheme());
+      setGlobal(NotificationService, NotificationService());
+      setGlobal(
+        PreferencesController,
+        preferencesController = PreferencesController(),
+      );
+    });
+
+    testWidgetsWithWindowSize('settings update preferences', windowSize,
+        (WidgetTester tester) async {
+      await pumpMemoryScreen(tester);
+      _setUpServiceManagerForMemory();
```

Comment:
this should be done in `setUp`, and it appears that there is a bit of overlap between `_setUpServiceManagerForMemory` and the `setUp` method above. Please get rid of the overlap. Better yet, there doesn't appear to be a reason for having two methods at all, so doing all the set up tasks directly in the `setUp` test lifecycle method is best.